### ISSUE TITLE
no-unused-vars false positive when interface and class have same name

### ISSUE THREAD

**Issue by glensc** [eslint/eslint#13146]
**Tell us about your environment**

* **ESLint Version:** v6.8.0
* **Node Version:** v12.16.1
* **npm Version:** (using yarn)

**What parser (default, Babel-ESLint, etc.) are you using?**

- @typescript-eslint/parser

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```yaml
env:
  es6: true
  node: true

extends:
  - 'eslint:recommended'
  - 'plugin:@typescript-eslint/eslint-recommended'

globals:
  Atomics: readonly
  SharedArrayBuffer: readonly

parser: '@typescript-eslint/parser'
parserOptions:
  ecmaVersion: 2018
  sourceType: module

plugins:
  - '@typescript-eslint'

rules:
  indent:
    - error
    - 2
  linebreak-style:
    - error
    - unix
  quotes:
    - error
    - double
  semi:
    - error
    - always

# vim:ts=2:sw=2:et

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```ts
import { EventEmitter } from "events";

const blah = (emitter: EventEmitter) => {
  emitter.on("start", () => {
  });
};
console.log(blah);

```

```bash
$ eslint src/eslint.ts
```

**What did you expect to happen?**

The unused var report should not be emitted

**What actually happened? Please include the actual, raw output from ESLint.**

```
src/eslint.ts
  1:10  error  'EventEmitter' is defined but never used  no-unused-vars

✖ 1 problem (1 error, 0 warnings)
```

**Are you willing to submit a pull request to fix this bug?**

If I knew how to fix it, I would have submitted it already.

**Comment by anikethsaha** [comment]
i think it should be reported in `typescript-eslint`
also, make sure you have turn off the base `no-unused-vars` (eslint) and using the `typescript-eslint/no-unused-vars` . though it is done by the recommended config. 

I think you need to use [`no-unused-vars-experimental`](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-unused-vars-experimental.md) instead

**Comment by mdjermanovic** [comment]
@glensc thanks for the issue!

As @anikethsaha already noted, for typescript code you should probably turn off the core `no-unused-vars` rule and use a corresponding rule from `@typescript-eslint/eslint-plugin` instead, which would be either [`@typescript-eslint/no-unused-vars`](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-unused-vars.md) or [`@typescript-eslint/no-unused-vars-experimental`](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-unused-vars-experimental.md).

**Comment by glensc** [comment]
confirmed, thanks!
